src/database/database.py

```
@@ -159,10 +159,30 @@ async def initialize(self) -> bool:
                     log.info("Inserted initial schema version.")
 
-                # Commit all changes
-                await conn.commit()
+                # Check current version and apply any pending migrations
+                cursor = await conn.execute("SELECT MAX(version) as version FROM schema_version")
+                row = await cursor.fetchone()
+                current_version = row[0] if row and row[0] is not None else 0
+
+                # Get all migration methods
+                migration_methods = [
+                    (version, method) for name, method in self.__class__.__dict__.items()
```

Ruff (0.8.2)

Undefined name `version`

(F821)